src/Types.js

```diff
@@ -53,8 +53,10 @@ class Types {
         });
     }
 
-    constructor(deserialize: (value: any) => any) {
-        this.deserialize = deserialize;
+    constructor(deserializer: (value: any) => any) {
+        this.deserialize = (value: any) =>
+            deserializer(value === null ? this.defaultValue() : value);
```

Comment:
We probably don't need to deserialize the default value. It's the developer's responsibility to ensure the default value matches the expected type. So `value == null ? this.defaultValue() : deserialize(value)` should be enough. The `==` here is useful to match both `undefined` and `null` (`=== null` returns false for `undefined`).

Reply:
As the Types are extendables, calling deserializer everytime is probably the right thing as you made need to process the default value as well ( ex: default date as string should still be parsed as date obj.). This way someone can add a type and handle the default value it's own way.

Reply:
That makes perfect sense! 👍